addons/binding/org.openhab.binding.velbus/src/main/java/org/openhab/binding/velbus/handler/VelbusThingHandler.java

```diff
@@ -0,0 +1,174 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.velbus.handler;
+
+import static org.openhab.binding.velbus.VelbusBindingConstants.*;
+
+import java.lang.invoke.MethodHandles;
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.thing.binding.builder.ChannelBuilder;
+import org.eclipse.smarthome.core.thing.binding.builder.ThingBuilder;
+import org.eclipse.smarthome.core.thing.type.ChannelKind;
+import org.eclipse.smarthome.core.thing.type.ChannelTypeUID;
+import org.openhab.binding.velbus.internal.VelbusModuleAddress;
+import org.openhab.binding.velbus.internal.VelbusPacketListener;
+import org.openhab.binding.velbus.internal.packets.VelbusStatusRequestPacket;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Base ThingHandler for all Velbus handlers.
+ *
+ * @author Cedric Boon - Initial contribution
+ */
+public abstract class VelbusThingHandler extends BaseThingHandler implements VelbusPacketListener {
+    /** Logger Instance */
+    protected final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+    private VelbusBridgeHandler velbusBridgeHandler;
+    private VelbusModuleAddress velbusModuleAddress;
+    private int numberOfChannels;
+    private String acceptedItemType;
+
+    public VelbusThingHandler(Thing thing, int numberOfChannels, int numberOfSubAddresses, String acceptedItemType) {
+        super(thing);
+
+        this.velbusModuleAddress = createVelbusModuleAddress(thing, numberOfSubAddresses);
+        this.numberOfChannels = numberOfChannels;
+        this.acceptedItemType = acceptedItemType;
+    }
+
+    private VelbusModuleAddress createVelbusModuleAddress(Thing thing, int numberOfSubAddresses) {
+        byte address = hexToByte((String) getConfig().get(MODULE_ADDRESS));
```

Comment:
The address config parameter on the thing is specified as "required", is it still possible that getConfig().get(MODULE_ADDRESS) returns null?

Reply:
Until your handler is initialized this parameter can indeed be `null`. So I would suggest to move your code from the constructor into the `initialize` method. The framework ensures that all required parameters are set before `initialize` is called on the handler.

While we are at this conversion anyway: I think it would make sense to use the [Hexutil](https://github.com/eclipse/smarthome/blob/master/bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/util/HexUtils.java#L114) to do the conversion.

Reply:
I moved the call to createVelbusModuleAddress(...) from the constructor to the initializeThing(...) method, and replaced the custom parsing of the hex string by Hexutil.